### ISSUE TITLE
ArgumentError: Error #3672: Buffer creation failed. Internal error

### ISSUE THREAD

**Issue by jenschr** [Gamua/Starling-Framework#267]
Whenever I turn on BlurFilter.createDropShadow in my app, I will get the above error after some time. The amount of time will vary, but it will eventually always happen. I can't easily reproduce this, but I'm using BlurFilter.createDropShadow for adding a drop shadow to text fields. I've attached a picture of the stack-trace since github won't allow me to attach a text file(?).

Any idea what happens?
![Screen Shot 2013-01-19 at 10 53 23 PM](https://f.cloud.github.com/assets/550202/80686/27d0f246-6283-11e2-81f3-9c07366ef7c0.png)

**Comment by scottenders** [comment]
I am unable to get a simple hello world app going on my phone without this error. Am I basically screwed until Adobe fixes that bug?

**Comment by PrimaryFeather** [comment]
@scottenders I haven't seen this happening in a simple Hello World app. Which AIR version are you using? I've heard from some problems with AIR 17 on specific Android devices, perhaps your issue is connected?

**Comment by PrimaryFeather** [comment]
(I'm thinking about [this issue](https://github.com/Gamua/Starling-Framework/issues/676#issuecomment-82400698) in particular.)

**Comment by scottenders** [comment]
OS: **Windows 7**
Flex SDK: **4.6.0**
AIR SDK: **16.0.0.292**
Starling: **1.6**
Phone: **Motorola Droid Razr**

Error:
```Exception fault: ArgumentError: Error #3672: Buffer creation failed. Internal error.
	at flash.display3D::Context3D/createVertexBuffer()
	at starling.display::QuadBatch/createBuffers()[/Users/redge/Dropbox/Development/starling/starling/src/starling/display/QuadBatch.as:174]
	at starling.display::QuadBatch/syncBuffers()[/Users/redge/Dropbox/Development/starling/starling/src/starling/display/QuadBatch.as:203]
	at starling.display::QuadBatch/renderCustom()[/Users/redge/Dropbox/Development/starling/starling/src/starling/display/QuadBatch.as:221]
	at starling.core::RenderSupport/finishQuadBatch()[/Users/redge/Dropbox/Development/starling/starling/src/starling/core/RenderSupport.as:501]
	at starling.core::Starling/render()[/Users/redge/Dropbox/Development/starling/starling/src/starling/core/Starling.as:515]
	at starling.core::Starling/nextFrame()[/Users/redge/Dropbox/Development/starling/starling/src/starling/core/Starling.as:469]
	at starling.core::Starling/onEnterFrame()[/Users/redge/Dropbox/Development/starling/starling/src/starling/core/Starling.as:682]```

**Comment by PrimaryFeather** [comment]
Okay, thanks! And which Android version?

**Comment by scottenders** [comment]
Android: **4.1.2**

**Comment by PrimaryFeather** [comment]
Adobe is already looking into it — thanks for your report!

**Comment by robot04** [comment]
I have the same error with Adobe AIR Beta 24

**Comment by PrimaryFeather** [comment]
Is it easily reproducible? In that case, please create an issue here: [tracker.adobe.com](https://tracker.adobe.com/), attach a minimal sample project, and post the link to the report here, so that we can vote for it. Don't forget to note down exactly on which device the error occurs. 

Thanks in advance!

**Comment by robot04** [comment]
Hi, Daniel. 

It seems to be solved in the next update of Adobe AIR 24 beta.
https://forums.adobe.com/thread/2229692

Thanks.

**Comment by PrimaryFeather** [comment]
I'm glad to hear that! Thanks for the update.

**Comment by manno-xx** [comment]
Mmmm, It seems not solved yet, I am seeing this too. Worked OK (and still does) in what I figure was last compiled with AIR SDK 22.

The offending lines are:

var cmFilter:ColorMatrixFilter = new ColorMatrixFilter();
cmFilter.tint( LINE_COLORS[ (this.option as ToneAnswerOptionVO).line ], 1 );
...
image.filter = cmFilter;

Commenting those lines shows the images OK in B&W.

The ATF texture atlas that contains images with the filter applied is 1024 x 256. Actual textures are at max 3 different ones of which the largest is 100x100. There's 6 ocurrences in total on the stage. The only other atlas' is a standard feathers one and one of 1024 x 512. There is hardly any gfx on screen when it happens

It is the app's first version on iPone. I have not published the new version with an old SDK but 'iPhone' itself cannot be the problem as it works on iPhone7 but not on iPad4 with the SDK 24 but worked on the iPad with SDK 22.

Some stats:

Starling 1.6
Feathers 2.0.1
(Yes, Starling and Feathers are old, was holding of the update to save precious time...)

AIR SDK 24.0.0.180

iPhone 7
iOS 10.2.1
Works OK

iPad 4
iOS 10.2 (Updating as we speak to see if that's the thing. Will post results)
Crashes

iPone 4
iOS ? (still have to check...)
Crashes

iPhone 5
iOS 10.2
Crashes

**Comment by manno-xx** [comment]
iPad 4 iOS 10.2.1 crashes too :(

**Comment by PrimaryFeather** [comment]
Does the [AIR 25 Beta](http://labs.adobe.com/downloads/air.html) change anything, perhaps?

**Comment by manno-xx** [comment]
Will check today or tomorrow and report. (did not see a mention in the release notes)

**Comment by manno-xx** [comment]
Sadly no, the 25.0.0 108 beta does not fix this issue. Same result: ArgumentError: Error #3672: Buffer creation failed. Internal error.

**Comment by cgascons** [comment]
There's a guy here (https://forums.adobe.com/thread/2229692?start=40&tstart=0) claiming this issue to be fixed AIR SDK Beta 25.0.0 build 118. I haven't yet found the time to test it myself but I thought I'd let you know about this.

**Comment by OMA2k** [comment]
Hello. I'm having this #3672 internal error message even with the latest AIR 25.0.0.180 published yesterday. But in my case it's not when using filters. It's when using VideoTexture in combination with drawToBitmapData(). This problem does NOT happen in AIR 24. Only AIR 25 betas (both AIR 25 beta builds 108 and 180). I'm not sure if I can reproduce the problem in an empty app, though. I'll try.

**Comment by manno-xx** [comment]
Thanks for the heads up cgascons. No such luck however :( 

But: build 180? I currently (feb 17th) only get build 118...

**Comment by sarahnorthway** [comment]
Confirmed the issue is fixed for me in AIR 25 beta.

I was getting #3672 errors on iPhone 4s and other older iOS hardware at starling.display::QuadBatch/createBuffers() when calling RenderTexture.drawBundled. Broken in AIR 24, works in AIR 22 and AIR 25 beta.

**Comment by cgascons** [comment]
Hey @manno-xx sorry for the very late reply didn't get the mention to the email because of the missing '@'  It looks like it's definitely not working yet, we are getting some of these errors as well on our Flash version of the games (browser) running AIR 25 (build 143).

**Comment by alevys** [comment]
Hi! I get this error when use Context3DProfile.BASELINE_CONSTRAINED in the emulator on Macbook. I use the extension for the Starling-Extension-Graphics when you draw a line. Error occurs when closing the app. If you use the auto profile, then everything works fine. I use AIR 27.0.0.128

**Comment by serjek** [comment]
confirmed in AIR 28 as well

**Comment by roipeker** [comment]
```
ArgumentError: Error #3672: Buffer creation failed. Internal error.
	at flash.display3D::Context3D/createVertexBuffer()
	at starling.rendering::VertexData/createVertexBuffer()[..\rendering\VertexData.as:904]
```
	
VM Windows 10 on Parallels Desktop (Macbook) running latest Starling commits (20/03/18) and lateest AIR SDK 29.0.0.120.

Basic "empty" project (just the main class) using "auto" render mode and profile.

AIR SDK v26.0.0.118 is working properly.

Any advise?

**Comment by PrimaryFeather** [comment]
Issues that happen on Windows only > AIR 26 might be related to the switch to Direct3D 11. Please try what happens when you use the following compiler argument (using AIR 29): `-swf-version=37`

That way, it will stick to Direct3D 9.

**Comment by milestilmann** [comment]
I had to update a very old project to AIR SDK 29 today (recent windows 10 update breaks a lot of of AIR-based games on Steam) and I can confirm that I got that weird "buffer creation" error, and then I am happy to report that the "-swf-version=37" compiler flag fixes it.  Thanks @PrimaryFeather !

**Comment by PrimaryFeather** [comment]
Thanks for the feedback -- I'm glad to hear the workaround is helping! :smile:
I told Adobe about the problem, but for now this is an easy fix.

**Comment by PippoApps** [comment]
This still happens with AIR 33. It only happens on my laptop, not on the desktop.
I confirm with AIR 26 it works.
Would be great if Harman could fix this. 
In the meantime, since I do compile with Animate, any suggestion how to set the "-swf-version=37" with Animate?
Thanks

**Comment by PrimaryFeather** [comment]
That's unfortunate! Harman has already fixed several problems with DirectX 11, but apparently not this one.

 I must admit I don't know how to change that "swf-version" setting in Adobe Animate – perhaps you can ask this in the official [Adobe Animate Forum](https://community.adobe.com/t5/animate/bd-p/animate?page=1&sort=latest_replies&filter=all)!

**Comment by DrNeroCF** [comment]
I found this while using VMWare to run CS6 from my BootCamp drive on Big Sur.  Easiest way around it is to compile a 32 / 33 SDK, then export just the swf as Flash Player 26 and replace the primary swf in the .app folder.

Anyone know if there would be drawbacks to using Flash Player 26?  Seems like almost all the new features passed 19 or so were to add support for new mobile standards, honestly.

To rant a bit here... as primarily a Mac user, it never occurred to me that Animate could export a standalone captive runtime under Windows, as .app folders are how Windows sees macOS application bundles.  For literally years I thought I had to futz around with the command line to get exes out of Animate.  Don't know whether to feel stupid or furious, haha.

Windows did have an issue for me of drawing all bitmaps black after loading around 10 levels or so unless using uncompressed textures, I think that was fixed more recently, don't know if it was the FP version or the SDK version though.

Need a new thread for compiling knowledge on how to get an AIR game on Steam nowadays :D
